### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by Zauberbutter**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for creating a TypeScript definition. 🙌 Can you follow [this styleguide](https://github.com/sindresorhus/typescript-definition-style-guide)?

Some examples:
- https://github.com/sindresorhus/semver-regex/commit/039944bb5b3876913171b895488874b5c200e458
- https://github.com/sindresorhus/global-dirs/commit/0aae4d64887f77af94c4fb7ca12aa9f93c0a8291
- https://github.com/sindresorhus/md5-hex/commit/bbbfeff840f26973fbffc16d4863ae6d120079f3

**Comment by Zauberbutter** [comment]
I didn't know this styleguide exists, I'll update this PR.

File: package.json [hunk 1]
```diff
@@ -33,6 +34,12 @@
...
 	"devDependencies": {
 		"ava": "^3.15.0",
-		"xo": "^0.39.1"
+		"xo": "^0.39.1",
+		"tsd": "^0.14.0"
+	},
+	"xo": {
+		"ignores": [
+			"*.ts"
+		]
```
Comment (sindresorhus) [inline package.json]: This should be removed. Linting should apply to the TS files too.

File: index.test-d.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import {expectType} from 'tsd';
+import truncateUrl from '.';
```
Comment (sindresorhus) [inline index.test-d.ts]:
```suggestion
import truncateUrl from './index.js';
```

**Pull request merged**